### ISSUE TITLE
Add a Pin button to the Inspector dock

### ISSUE THREAD

**Issue by skub007** [godotengine/godot-proposals#9265]
- *Production edit: Related to https://github.com/godotengine/godot-proposals/issues/9218.*

### Describe the project you are working on

A Cube game

### Describe the problem or limitation you are having in your project

i have this array to which i wanna add my obstacles to but when i select my obstacles the inspector would switgch to the obstaclwes not allowing me to do it

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

a lock inspector option which locks the inspector to a specific node which we choose like we choose the node their is a option (a lock icon) and we click it it locks so if we click any other node it would change inspector until we click it again to unlock it

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

allows to feed nodes to Node[] easily

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Maybe it will be used often

### Is there a reason why this should be core and not an add-on in the asset library?

....i didnt find any add-on

**Comment by AThousandShips** [comment]
I don't see why you would select another node while editing a node, why do you need this? Why do you need to be able to select another node *and* edit another?

**Comment by skub007** [comment]
> I don't see why you would select another node while editing a node, why do you need this? Why do you need to be able to select another node _and_ edit another?

i wanna add my obstacles nodes (there are many) to my node array in my player ( i made a node[] obstacles in c#) so if i drag and drop all the nodes to the array it would add all of them in thats why like i have my player who has array of nodes what i wanna do is select all the obstacles nodes and drag them on the node[] in the editor of the player (inspector panel for player) thus adding them in but when i click on any obstacle it would switch to it not allowing me to do it

**Comment by skub007** [comment]
[Screencast from 2024-03-08 19-31-05.webm](https://github.com/godotengine/godot-proposals/assets/161951708/ed9b1862-fb30-4573-ba07-6219ffb0df35)

**Comment by skub007** [comment]
![image](https://github.com/godotengine/godot-proposals/assets/161951708/244aec3d-e6a4-45af-acc1-bd19e88e0bcb)

**Comment by Mickeon** [comment]
> allows to feed nodes to Node[] easily

This feels like an issue with exported Array of Nodes themselves, and I do relate to this. A theoretical Lock button would only be a band-aid solution for this.

**Comment by Zireael07** [comment]
A bandaid which seems needed, though (had a similar problem when adding an array of nodes to my AI's path exported variable)

**Comment by AThousandShips** [comment]
I disagree that it's needed, if it has no other value than covering this limitation it's not a good addition

**Comment by RedMser** [comment]
Blender has a similar feature in the Properties editor, in the form of a "Pin" button. Though it's more useful there, as you can open multiple Properties editors at once (see #7459), so you can e.g. use it to compare values between two objects, or keep an editor for "global settings" open at all times.

**Comment by skub007** [comment]
> I disagree that it's needed, if it has no other value than covering this limitation it's not a good addition

what?! u think its niot needed?!?!?! locking inspector for editing its value while observing other node is not needed?! adding bunch of nodes to a node array at ones to save time and effort is not needed?!?!?! are you serious????!

**Comment by skub007** [comment]
godot is open-source that means needed things should be there and i have seen atleast 2 more people are saying it should be there yet you say its not needed?! what does open-source mean?!?!?!

**Comment by skub007** [comment]
i am not able to0 find such a addo-on online thats why i came here plus...i dont know how to add add-on as am new to godot and started just few weeks ago

**Comment by AThousandShips** [comment]
Please read the [code of conduct](https://godotengine.org/code-of-conduct) and behave professionally, you will find the answers to your questions about how this project is managed there :)

**Comment by AeioMuch** [comment]
> > allows to feed nodes to Node[] easily
> 
> This feels like an issue with exported Array of Nodes themselves, and I do relate to this. A theoretical Lock button would only be a band-aid solution for this.

Would be better to have a pick button next to the Array of nodes that allow you to select multiples nodes in the popup window and add them in batch indeed.

**Comment by skub007** [comment]
> > > allows to feed nodes to Node[] easily
> > 
> > 
> > This feels like an issue with exported Array of Nodes themselves, and I do relate to this. A theoretical Lock button would only be a band-aid solution for this.
> 
> Would be better to have a pick button next to the array of nodes that allow you to select multiples nodes in the popup window and add them in batch indeed.

that would also work for me litrellay any similer feasture that would make life easy...btw out of context dont take it seriously but why doesnt godot ship a universal installer? like for windows or linux..i understand that on linux its hard but for windows??

**Comment by AThousandShips** [comment]
See:
* https://github.com/godotengine/godot-proposals/issues/1432

Please keep things on topic, if you want some other features please search or open issues for it

**Comment by SeanRamey** [comment]
Yeah, I just wanted to drop in and say that all you really need to fix this issue is to add support for drag and dropping multiple nodes at once into things like and Array of NodePaths

https://github.com/godotengine/godot-proposals/assets/3057229/da7763dd-5bae-45a6-a9de-6a76b819973d

**Comment by Calinou** [comment]
@SeanRamey That particular feature is being tracked in https://github.com/godotengine/godot-proposals/issues/8193 already.

**Comment by skub007** [comment]
or godot 4 goes in lts

**Comment by theraot** [comment]
Please reopen this.

This is closed by a PR that does not add a pin button to the inspector (i.e it does not address this proposal):

- https://github.com/godotengine/godot/pull/98074

And now that the camera preview is in the inspector we have a new issue: With the camera in the inspector, instead of it being a separate dock, when you unselect the camera the preview goes away, which is an issue if you want to place the camera, and then move objects into frame. Being able to pin the inspector would address this.

**Comment by AeioMuch** [comment]
> And now that the camera preview is in the inspector we have a new issue: With the camera in the inspector, instead of it being a separate dock, when you unselect the camera the preview goes away, which is an issue if you want to place the camera, and then move objects into frame. Being able to pin the inspector would address this.

When you select the camera node, you have a `Preview` button in the Viewport as well. And you can split the Viewport into multiples via the `View` menu in the toolbar, to achieve what you want.

**Comment by theraot** [comment]
@AeioMuch I have an addon to achieve what I want. I still believe this proposal is wrongly closed, and that it is desirable.

**Comment by fire** [comment]
@theraot can you link to it? I'm still interesting is salvaging implementations of pinned inspectors.

**Comment by theraot** [comment]
I don't have an addon that pins the inspector, I have an addon that adds the camera preview as a new dock. Sadly that is not the path that Godot took. And I have not published it... I do not like that for Godot asset library it is an addon for repository, because I would have to make too many repositories. @fire

**Comment by JacobWMills** [comment]
Bonkers view that this feature isn't needed. As a designer the ability to have two Inspectors open would be super valuable. To compare similar nodes, to select and drag other objects into it without it changing, etc. Unity does it, Blender, etc. etc. etc.

**Comment by Mickeon** [comment]
I have read enough. I am reopening the proposal. It's true that https://github.com/godotengine/godot/pull/98074 adds a Pin button, but it's for the Bottom Panel, **not** for the Inspector dock. This proposal was not addressed.

**Comment by Calinou** [comment]
@BobbeDev Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by KoBeWi** [comment]
Closing in favor of #13684, which is more detailed.
